tests/test_barostat.py

```diff
@@ -167,15 +167,12 @@ def test_barostat_varying_pressure():
     baro.set_pressure((1 * unit.atmosphere).value_in_unit(unit.bar))
     # Changing the barostat interval resets the barostat step.
     baro.set_interval(2)
-    start = time.time()
-    volumes = []
-    while time.time() - start < 15:
-        ctxt.multiple_steps(np.ones(1000)*lam)
-        atm_box = ctxt.get_box()
-        volumes.append(compute_box_volume(atm_box))
 
+    ctxt.multiple_steps(np.ones(5000)*lam)
+    atm_box = ctxt.get_box()
+    np.testing.assert_equal(compute_box_volume(atm_box), 7.838565243228307)
```

Comment:
we probably don't want to make it *this* precise, because if we run the test on a different driver/GPU the numbers will be off. CUB only guarantees local-determinism (w.r.t. GPU/driver combo)

Reply:
Looks like even between T4s, this can differ by 0.4 which is more significant than I had expected...

Reply:
hm... it looks like you're doing atomicAdds actually, I think even with atomicAdds we can only guarantee run-to-run determinism.

Reply:
Have you confirmed that its deterministic on the same T4 instance?

Reply:
It was deterministic on a T4 and on my RTX 2080, both passing the tests.